### PR TITLE
Grpc api def

### PR DESCRIPTION
**Pull request by ryan-mcgregor**

### CODE REVIEW

File: internal/server/healthService.go [hunk 1]
```diff
@@ -103,5 +103,47 @@ func (h *HealthService) GetMentalHealthScoreForUser(
...
+) (*pbhealth.DeleteHealthDataForUserResponse, error) {
+	var err error
+	var numDeleted uint32
+
+	switch x := req.Data.(type) {
+	case *pbhealth.DeleteHealthDataForUserRequest_All:
+		numDeleted, err = h.db.DeleteMentalHealthLogs(ctx, req.UserID, nil, x.All)
+		break
+	case *pbhealth.DeleteHealthDataForUserRequest_DateToRemove:
+		numDeleted, err = h.db.DeleteMentalHealthLogs(ctx, req.UserID, x.DateToRemove, true)
```
Comment (regnivon) [inline internal/server/healthService.go]: This looks like a bug to me, you are saying "true" to all, but this is meant to be the case where we delete a specific day.

File: pkg/database/mongoRepository.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.uber.org/zap"
 	"log"
+	"math"
 	_ "strings"
```
Comment (regnivon) [inline pkg/database/mongoRepository.go]: What does this anonymous strings import do? Is it actually needed for something?

File: pkg/database/mongoRepository.go [hunk 3]
```diff
@@ -38,7 +39,7 @@ func (m *MongoRepository) SetCollections(databaseName string) {
 func (m *MongoRepository) AddMentalHealthLog(ctx context.Context, healthLog *pbhealth.MentalHealthLog) (string, error) {
 	res, err := m.fileCollection.InsertOne(context.TODO(), healthLog)
 	if err != nil {
-		m.logger.Infof("%v", err)
+		m.logger.Infof("Error adding mental health log: %v", err)
```
Comment (regnivon) [inline pkg/database/mongoRepository.go]: This should be Errorf not Infof

File: pkg/database/mongoRepository.go [hunk 4]
```diff
@@ -94,8 +95,10 @@ func (m *MongoRepository) GetAllMentalHealthLogsByDate(ctx context.Context, user
 	return toReturn, err
 }
 
-func (m *MongoRepository) GetOverallScore(ctx context.Context, userID int64) (float64, error) {
+func (m *MongoRepository) GetOverallScore(ctx context.Context, userID int64) (int32, error) {
 	logs, err := m.GetAllMentalHealthLogs(ctx, userID)
+	log.Printf("Logs fetched for user (ID = %v\n): %v\n", userID, logs)
```
Comment (regnivon) [inline pkg/database/mongoRepository.go]: Use the injected zap logger stored in the struct, not the stdlib logger.

File: pkg/database/mongoRepository.go [hunk 5]
```diff
@@ -94,8 +95,10 @@ func (m *MongoRepository) GetAllMentalHealthLogsByDate(ctx context.Context, user
...
+
 	if err != nil {
 		log.Fatal("cannot get mental health logs for user: ", err)
```
Comment (regnivon) [inline pkg/database/mongoRepository.go]: Use the injected zap logger stored in the struct, not the stdlib logger. (This is the case in many points here, resolve them all please)

File: pkg/database/mongoRepository.go [hunk 6]
```diff
@@ -109,10 +112,57 @@ func (m *MongoRepository) GetOverallScore(ctx context.Context, userID int64) (fl
...
+	if all {
+		filter = bson.M{"userid": userID} // filtering by user id and date
+	} else {
+		filter = bson.M{"userid": userID, "logdate": date} // filtering by user id and date
+	}
+
+	res, err := m.fileCollection.DeleteMany(ctx, filter) // deleting all health logs with the given date
+
+	if err != nil {
+		log.Fatal("cannot delete mental health logs for user: ", err)
```
Comment (regnivon) [inline pkg/database/mongoRepository.go]: Do not use Fatal here!!! The server can very easily recover from not deleting mental health logs!! This should be a log to ErrorF and then return the error to the caller to be handled

File: pkg/database/mongoRepository.go [hunk 7]
```diff
@@ -109,10 +112,57 @@ func (m *MongoRepository) GetOverallScore(ctx context.Context, userID int64) (fl
...
+	update := bson.M{
+		"$set": healthLog,
+	}
+
+	_, err := m.fileCollection.UpdateOne(
+		ctx,
+		filter,
+		update)
+	if err != nil {
+		m.logger.Infof("Error updating mentalh health log: %v", err)
```
Comment (regnivon) [inline pkg/database/mongoRepository.go]: Errorf and then return error to caller

File: internal/server/healthService.go [hunk 8]
```diff
@@ -103,5 +103,47 @@ func (h *HealthService) GetMentalHealthScoreForUser(
...
+}
+
+func (h *HealthService) UpdateHealthDataForDate(
+	ctx context.Context,
+	req *pbhealth.UpdateHealthDataForDateRequest,
+) (*pbhealth.UpdateHealthDataForDateResponse, error) {
+
+	err := h.db.UpdateMentalHealthLogs(ctx, req.UserID, req.DesiredLogInfo)
+	if err != nil {
+		h.logger.Infof("%v", err)
```
Comment (regnivon) [inline internal/server/healthService.go]: errorf

File: internal/server/healthService.go [hunk 9]
```diff
@@ -103,5 +103,47 @@ func (h *HealthService) GetMentalHealthScoreForUser(
...
+		return &pbhealth.UpdateHealthDataForDateResponse{
+			Success: false,
+		}, status.Errorf(codes.Internal, "Error updating mental health logs")
```
Comment (regnivon) [inline internal/server/healthService.go]: Probably should use something other than the internal error code here, perhaps invalid arguments or something

**Review by regnivon — CHANGES_REQUESTED** [review]

File: internal/server/healthService.go (hunk 1 above)
Comment (ryan-mcgregor) [reply internal/server/healthService.go]: Good catch!

File: pkg/database/mongoRepository.go (hunk 2 above)
Comment (ryan-mcgregor) [reply pkg/database/mongoRepository.go]: I used math.round() to round an integer, so this is needed

Comment (ryan-mcgregor) [reply pkg/database/mongoRepository.go]: Oh, you're right! Good catch!

File: pkg/database/mongoRepository.go (hunk 3 above)
Comment (ryan-mcgregor) [reply pkg/database/mongoRepository.go]: Changed!

File: pkg/database/mongoRepository.go (hunk 5 above)
Comment (ryan-mcgregor) [reply pkg/database/mongoRepository.go]: Changed all the instances of this!

File: pkg/database/mongoRepository.go (hunk 6 above)
Comment (ryan-mcgregor) [reply pkg/database/mongoRepository.go]: Understood, thank you!

File: pkg/database/mongoRepository.go (hunk 7 above)
Comment (ryan-mcgregor) [reply pkg/database/mongoRepository.go]: Changed, good catch!

File: internal/server/healthService.go (hunk 8 above)
Comment (ryan-mcgregor) [reply internal/server/healthService.go]: Changed

File: internal/server/healthService.go (hunk 9 above)
Comment (ryan-mcgregor) [reply internal/server/healthService.go]: Changed it to Invalid Argument. Thanks!

**Review by regnivon — APPROVED** [review]

**Pull request merged**